### PR TITLE
HAAR-533: Cleaning up user search code

### PR DESCRIPTION
**Pull request by brightonsbox**

### CODE REVIEW

File: backend/controllers/searchDpsUsers.js [hunk 1]
```diff
@@ -5,25 +5,24 @@ const size = 20
 
 const searchFactory = (
   paginationService,
-  getAssignableGroupsOrPrisonsApi,
```
Comment (brightonsbox) [inline backend/controllers/searchDpsUsers.js:8]: Now prison specific for DPS search

File: backend/controllers/searchDpsUsers.js [hunk 2]
```diff
@@ -51,14 +50,12 @@ const searchFactory = (
...
 
-    res.render('searchWithFilter.njk', {
+    res.render('searchDpsUsers.njk', {
       searchTitle,
       searchUrl,
-      groupOrPrisonDropdownValues,
+      prisonDropdownValues,
       roleDropdownValues,
-      showGroupOrPrisonDropdown,
-      dpsSearch,
```
Comment (brightonsbox) [inline backend/controllers/searchDpsUsers.js:60]: Don't need to specify if DPS search or not

File: backend/controllers/searchDpsUsers.js [hunk 3]
```diff
@@ -51,14 +50,12 @@ const searchFactory = (
...
-      groupOrPrison: dpsSearch ? 'caseload' : 'group',
```
Comment (brightonsbox) [inline backend/controllers/searchDpsUsers.js:61]: Don't need alternate wording, will always be caseload

File: backend/controllers/searchDpsUsers.test.js [hunk 4]
```diff
@@ -22,7 +22,6 @@ describe('search factory', () => {
       '/search-with-filter-dps-users',
       '/manage-dps-users',
       'Search for a DPS user',
-      true,
```
Comment (brightonsbox) [inline backend/controllers/searchDpsUsers.test.js:25]: Can remove dpsSearch flag here now

File: backend/controllers/searchExternalUsers.js [hunk 5]
```diff
@@ -5,40 +5,35 @@ const mapUsernameAndEmail = (u) => {
 
 const searchFactory = (
   paginationService,
-  getAssignableGroupsOrPrisonsApi,
```
Comment (brightonsbox) [inline backend/controllers/searchExternalUsers.js:8]: Now group specific for external user search

File: backend/controllers/searchExternalUsers.js [hunk 6]
```diff
@@ -5,40 +5,35 @@ const mapUsernameAndEmail = (u) => {
...
-    const showGroupOrPrisonDropdown = Boolean(res.locals?.user?.maintainAccessAdmin || !dpsSearch)
 
-    res.render('search.njk', {
+    res.render('searchExternalUsers.njk', {
       searchTitle,
       searchUrl,
-      groupOrPrisonDropdownValues,
+      groupDropdownValues,
       roleDropdownValues,
-      showGroupOrPrisonDropdown,
```
Comment (brightonsbox) [inline backend/controllers/searchExternalUsers.js:34]: Always show group dropdown values for external search (code used to be: `const showGroupOrPrisonDropdown = Boolean(res.locals?.user?.maintainAccessAdmin || !dpsSearch)`)

File: backend/controllers/searchExternalUsers.test.js [hunk 7]
```diff
@@ -0,0 +1,247 @@
+const { searchFactory } = require('./searchExternalUsers')
```
Comment (brightonsbox) [inline backend/controllers/searchExternalUsers.test.js:1]: Just the search.test.js file renamed, and a few bits changed to match the variables passed to nunjucks

File: backend/routes.js [hunk 8]
```diff
@@ -25,10 +25,7 @@ const configureRoutes = ({ oauthApi, manageUsersApi, nomisUsersAndRolesApi }) =>
   router.use('/create-dps-user', createDpsUserRouter({ nomisUsersAndRolesApi, manageUsersApi }))
   router.use('/create-external-user', createAuthUserRouter({ oauthApi }))
   router.use('/search-external-users', searchExternalUserRouter({ oauthApi }))
-  router.use(
-    '/search-with-filter-dps-users',
-    searchWithFilterDpsUserRouter({ oauthApi, nomisUsersAndRolesApi, manageUsersApi }),
-  )
+  router.use('/search-with-filter-dps-users', searchDpsUserRouter({ oauthApi, nomisUsersAndRolesApi, manageUsersApi }))
```
Comment (brightonsbox) [inline backend/routes.js:28]: I wonder if we can change the URL too? Or will people have bookmarked this?

**Review by Mjwillis — APPROVED** [review]
LGTM

**Review by SimonMitchellMOJ — APPROVED** [review]
LGTM

**Pull request merged**